### PR TITLE
fix: support PHP 8

### PR DESCRIPTION
**Pull request by chris-doehring**
Hey there,

your package is a dependency for many other packages and therefor it would be nice to allow php 8. There is only one change that I would classify as "potentially critical" in order to have a running phpstan analysis: I had to add a default behaviour to the switch case of `LanguageServerProtocol\CompletionItemKind::fromSymbolKind`, which is indeed required, as the return type hint only allows integer values.

Otherwise travis CI runs without problems under php 7.0, 7.3, 7.4 and 8.0-dev.

### CODE REVIEW

File: src/CompletionItemKind.php
Comment (chris-doehring) [inline src/CompletionItemKind.php:67]: The alternative would be to throw an exception as default behaviour:
```php
throw new \InvalidArgumentException('There is no SymbolKind with value "'.$kind.'"');
```

**Comment by chris-doehring** [comment]
Hey @felixfbecker, could you maybe check out this pull request when you got some time? Thanks in advance!

File: phpstan.neon
Comment (MidnightDesign) [inline phpstan.neon:2]: What's the reason behind this? Can't you just remove the ignored errors?

File: src/CodeLens.php
Comment (MidnightDesign) [inline src/CodeLens.php]: Those are redundant. As are some of the others you've added.

Comment (chris-doehring) [reply src/CodeLens.php]: Indeed, I will remove those two, but the one for `$data` is required as phpstan will otherwise throw: `Method LanguageServerProtocol\CodeLens::__construct() has parameter $data with no typehint specified.`

File: phpstan.neon
Comment (chris-doehring) [reply phpstan.neon:2]: The existing "ignoreErrors" options is helpful when running phpstan under php 7.0, as it would throw something like this otherwise:
`TextDocumentItem::$uri (string) does not accept string|null.`
Under php >=7.1 this does not happen anymore, but then you will get the following error instead:
` Ignored error pattern #Property .* does not accept# was not matched in reported errors.`
Which is the reason I have added the `reportUnmatchedIgnoredErrors: false` option.

Comment (MidnightDesign) [reply phpstan.neon:2]: Then I guess a better option would be to pin the PHP version for PHPStan. You can set it using `parameters.phpVersion: 70000` (for 7.0) in `phpstan.neon`.

File: src/CodeLens.php
Comment (MidnightDesign) [reply src/CodeLens.php]: Yup, that's why I highlighted just those two lines. : ) Thanks!

File: phpstan.neon
Comment (chris-doehring) [reply phpstan.neon:2]: But that doesn't work when running the analysis under `>=7.1`:
` The option 'parameters › phpVersion' expects to be int:70100..80000|null, 70000 given`

Or do you mean creating php version specific phpstan configuration files?

Comment (MidnightDesign) [reply phpstan.neon:2]: Oh, okay. I don't understand the problem _exactly_, but I just think there _has_ to be a better way. But you be the judge of that. : )

Comment (chris-doehring) [reply phpstan.neon:2]: Well, I agree that adding this option is a bit annoying, but it's only ignoring an error which will occur due to the nature of the different php versions and due to the one specific ignored error which is also defined in the same file. So I would say it’s the best solution to add it : )

**Comment by muglug** [comment]
@felixfbecker let me know if you want someone to help with maintenance of this repo – I know how gruelling open source can be

**Review by felixfbecker — APPROVED** [review]

File: .travis.yml
Comment (felixfbecker) [inline .travis.yml:7]: Is there an explicit version we could set here so that CI can't break for the next major PHP version?

Comment (chris-doehring) [reply .travis.yml:7]: At the time when I implemeted that, Travis did not support the handle `8.0`, only when using `nightly`. But it seems that this is still the case: https://travis-ci.community/t/php-8-0-missing/10132

Comment (chris-doehring) [reply .travis.yml:7]: With GitHub actions there is a way to do that, though.

Comment (felixfbecker) [reply .travis.yml:7]: Alright, doesn't need to block this PR then

**Comment by felixfbecker** [comment]
@muglug your comment hit a spot, thank you for the offer! I'm scared of handing out write access lightly but given you're maintaining Psalm I trust this is in good hands with you. Added you to the repo and feel free to not block on me for any changes. I just don't have time to dedicate to PHP anymore because I'm exclusively in the TypeScript world nowadays, but of course I'd still love to see these projects continue to be useful to the PHP community.

**Comment by muglug** [comment]
Thanks!

**Pull request merged**
